packages/react-combobox/src/utils/Selection.types.ts

```diff
@@ -2,7 +2,7 @@ import * as React from 'react';
 
 export type SelectedOption = {
   /** The `key` prop of the option. */
-  key: string;
+  id: string;
```

Comment:
Do I correctly understand that we are going to use DOM `id` as a unique identifiers for items?

Reply:
Sort-of. The `id` is what identifies an option for as long as it is mounted for registern/unregister and `activedescendant` stuff, but isn't stable across e.g. multiple opens/closes of the listbox. After this, I'll need to update `useSelection` to work by comparing the Combobox value to the option value/calculated value so the selection is persistent.

There are some nuances there where if an author wants uncontrolled selection to work, the value of options (either through the `value` prop, or calculated based on content) will need to be stable. The safest way to do that would be to define the `value` prop, so they wouldn't break things if they e.g. added an icon or extra text or something to the option children.